### DISCUSSION TITLE
Chunkserver on Master PC?

### DISCUSSION THREAD

**Discussion by Blackpaw** [moosefs/moosefs#424] [Q&A]
As per the subject - are there any issues with this?

**Comment by chogata** [comment]
No specific issues other than what common sense tells you: your PC needs to be able to carry both the master process and the chunkserver process, so enough RAM for both running without using swap space, good enough processor and, while the master process is single threaded apart from separate processes for saving metadata, chunkserver is multi threaded, so in general a different kind of processor is recommended for each of those (master - not many cores, but highest possible performance per single core, chunkserver - highest performance overall). It all depends on how busy your installation is, if not overly so, then the setup will not have any drawbacks perfomance wise. And of course remember the danger of having two pieces of your installation needing to be reinstalled/moved to another machine in case of any failure. 
Theoretically the processes could even share common local data directory (default: /var/lib/mfs), as they record files with different names, however, for clarity's sake, it's better to give them separate working directories. And if at all possible, do not use the same physical disk for chunks storage for the chunkserver and for metadata storage for the master.
If you have enough machines, it's cleaner to not keep the master and chunkserver together. But if your installation is small in terms of physical machine count and you could benefit from this one extra chunkserver (one more is always better for replication algorithms and data access speed), then this is overall not a bad idea.

**Reply by Blackpaw** [reply]
Thanks, the load is quite low - media server with a couple of clients, so mainly reads, lots of ram and cpu, underlying disk is zfs on a NVMe SSD, high endurance and speed.

I hadn't considered the sharing of local directories an issue. The mfs server is actually a LXD container, so I think I'll split the chunkserver out as a 2nd instance - better separation that way.

Thanks!
